### PR TITLE
add Assetto Corsa (Freeroam + AI) & Cleanup Assetto Corsa

### LINKED ISSUE #2052: [Egg Request]: Assetto Corsa (Freeroam + AI)
### Does this expand an already existing service

No

### Link to a game or software

https://assettoserver.org/

### Links for the download

https://github.com/compujuckel/AssettoServer/releases

### Links for the install docs

https://github.com/compujuckel/AssettoServer

### PR DESCRIPTION
**Pull request by QuintenQVD0**
# Description

- add Assetto Corsa (Freeroam + AI) & Cleanup Assetto Corsa
- closes #2052
- update normal egg to PTDL_v2
- full install script
- move to `^C`
- move to the steamcmd image

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

* [x] You verify that the start command applied does not use a shell script
  * [ ] If some script is needed then it is part of a current yolk or a PR to add one
* [x] The egg was exported from the panel

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa--freeroam--a-i.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            "container": "ghcr.io\/parkervcp\/installers:debian",
+            "entrypoint": "bash"
+        }
+    },
+    "variables": [
+        {
+            "name": "Version",
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa--freeroam--a-i.json:32]: no need for this variable, bcause only latest is supported

File: game_eggs/steamcmd_servers/assetto_corsa/README.md [hunk 2]
```diff
@@ -8,5 +8,8 @@ Asseto Corsa requires two ports, game port is UDP and WEB HTTP port is TCP.
 
 | Port        | default |
 |-------------|---------|
-| Game        | 9600   |
-| HTTP       | 8081   |
+| Game        | 9600    |
+| HTTP        | 8081    |
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/assetto_corsa/README.md:12]: you missed a notice, that every port can be used.

File: game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa.json [hunk 3]
```diff
@@ -109,7 +117,8 @@
             "default_value": "1",
             "user_viewable": true,
             "user_editable": true,
-            "rules": "required|string|max:1"
+            "rules": "required|boolean",
+            "field_type": "text"
         },
         {
             "name": "Windows param",
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa.json:124]: The official name is "WINDOWS_INSTALL"

File: game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa.json [hunk 4]
```diff
@@ -91,7 +97,8 @@
             "default_value": "8081",
```
Comment (gOOvER) [inline game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa.json:97]: dont provide a port, because this is part of the admin

**Review by gOOvER — CHANGES_REQUESTED** [review]

File: game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa.json (hunk 4 above)
Comment (QuintenQVD0) [reply game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa.json:97]: ?

File: game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa.json (hunk 3 above)
Comment (QuintenQVD0) [reply game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa.json:124]: End users do not see that?

File: game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa.json (hunk 4 above)
Comment (QuintenQVD0) [reply game_eggs/steamcmd_servers/assetto_corsa/egg-assetto-corsa.json]: I removed the default value and set it to nullable

**Comment by HerrSammyDE** [comment]
merge?

**Review by parkervcp — APPROVED** [review]

**Pull request merged**